### PR TITLE
[Model] Lda subgraph

### PR DESCRIPTION
**Pull request by yifeim**
## Description
Add subgraph dataloader support (see the example at the end of the model file). Revise README to better explain LDA details. Simplify code.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes
Besides refactoring, some changes are introduced to improve readability:

1. rename `word_rho` to `rho`
2. rename `step_size` to `mult`.

### CODE REVIEW

File: examples/pytorch/lda/lda_model.py [hunk 1]
```diff
@@ -98,133 +79,207 @@ class LatentDirichletAllocation:
...
+    model.sample(model.transform(G), 2)
     model.perplexity(G)
+
+    sampler = dgl.dataloading.MultiLayerFullNeighborSampler(1)
+    dataloader = dgl.dataloading.NodeDataLoader(
+        G.reverse(), {'doc': np.arange(G.num_nodes('doc'))}, sampler,
+        batch_size=1024, shuffle=True, drop_last=False)
+    for input_nodes, _, (block,) in dataloader:
+        B = dgl.DGLHeteroGraph(
+            block._graph, ['_', 'word', 'doc', '_'], block.etypes
```
Comment (BarclayII) [inline examples/pytorch/lda/lda_model.py:281]: Looks like this is constructing a normal graph from a block.  Does `dgl.block_to_graph` work for you?

Comment (yifeim) [reply examples/pytorch/lda/lda_model.py:281]: I wanted to only get "word_src->doc_dst" and then rename the nodes back to "word" and "doc". What is the best way to do that?

Comment (BarclayII) [reply examples/pytorch/lda/lda_model.py:281]: OK I just realized that there's not an easy way to rename a graph's node and edge types.  Though technically you could call `block_to_graph` and then reconstruct another graph with the same edges.  I'll leave it like this for now.

**Review by BarclayII — APPROVED** [review]

**Review by BarclayII — APPROVED** [review]

File: examples/pytorch/lda/lda_model.py (hunk 1 above)
Comment (yifeim) [reply examples/pytorch/lda/lda_model.py:281]: "call block_to_graph and then reconstruct another graph with the same edges" <- how?

**Comment by yifeim** [comment]
Btw @BarclayII I realized some better ways to improve numerical precision by using log_weight instead of plan weight. Please let me know if you want me to contribute in this PR or open another PR for the difference. Also, if you can show me how to rename the nodes, I could add that in as well.

**Comment by BarclayII** [comment]
> Btw @BarclayII I realized some better ways to improve numerical precision by using log_weight instead of plan weight. Please let me know if you want me to contribute in this PR or open another PR for the difference. Also, if you can show me how to rename the nodes, I could add that in as well.

Sure let's do that in this PR.  Let me know if you are done.

**Review by BarclayII — APPROVED** [review]

**Comment by BarclayII** [comment]
Thanks for your contribution!

**Pull request merged**